src/seedu/addressbook/data/person/Phone.java

```diff
@@ -6,54 +6,19 @@
  * Represents a Person's phone number in the address book.
  * Guarantees: immutable; is valid as declared in {@link #isValidPhone(String)}
  */
-public class Phone {
+public class Phone extends Contact{
 
-    public static final String EXAMPLE = "123456789";
-    public static final String MESSAGE_PHONE_CONSTRAINTS = "Person phone numbers should only contain numbers";
-    public static final String PHONE_VALIDATION_REGEX = "\\d+";
-
-    public final String value;
-    private boolean isPrivate;
-
-    /**
-     * Validates given phone number.
-     *
-     * @throws IllegalValueException if given phone string is invalid.
-     */
     public Phone(String phone, boolean isPrivate) throws IllegalValueException {
+    	
```

Comment:
Its unusual to throw exceptions from the constructor. It should be clear and clear for the purpose. Also, validation  of input values should be done at UI level